### PR TITLE
Notes API add endpoint to delete all notes for a user

### PR DESCRIPTION
**Pull request by yro**
### Description
Add 'delete all notes for user' endpoint for Notes API. Implements list view delete method.

### Testing
- [x] Unit tests

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @sanfordstudent 
- [ ] Code review: @doctoryes 

FYI: @edx/educator-neem

### Post-review
- [ ] Rebase and squash commits

### CODE REVIEW

File: notesapi/v1/views.py [hunk 1]
```diff
@@ -349,6 +348,19 @@ def post(self, *args, **kwargs):  # pylint: disable=unused-argument
         serializer = NoteSerializer(note)
         return Response(serializer.data, status=status.HTTP_201_CREATED, headers={'Location': location})
 
+    def delete(self, *args, **kwargs):  # pylint: disable=unused-argument
+        """
+        Delete all annotations for user_id
+
+        """
+        params = self.request.data
+        print(params)
```
Comment (sanfordstudent) [inline notesapi/v1/views.py]: Leftover print statement.

Comment (yro) [reply notesapi/v1/views.py]: 👍 Thank you.

**Comment by sanfordstudent** [comment]
How is auth handled for Notes? If Notes can only be called from the LMS, this all seems good to me. If Notes doesn't already have that auth in place though, we should add something as part of this PR. The other GDPR retirement endpoints are all restricted to superusers only. If only the LMS can call into Notes, that's fine, because we can put the same restrictions on the endpoint that calls this endpoint (the cascading endpoint thing we discussed in grooming would actually make sense).

**Comment by yro** [comment]
@sanfordstudent That's a really great question. There's no, or blisteringly out of date, documentation about this. I'll see if I can find something about it.

**Comment by yro** [comment]
@sanfordstudent It looks like there's a token process (at least) from the LMS, I will do some more investigation, but it appears to be moderately secure.

**Comment by yro** [comment]
@doctoryes Here's one of the PRs I was discussing with you yesterday. Thank you!

File: notesapi/v1/tests/test_views.py [hunk 2]
```diff
@@ -508,6 +507,40 @@ def test_pagination_page_start(self, page, start, previous_page, next_page):
...
+        response = self._get_search_results()
+        self.assertEqual(response["total"], 0)
+
+        # Reattempt delete for User 1
+        response = self.client.delete(url, self.headers)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Delete for No User
+        del self.headers["user_id"]
+        response = self.client.delete(url, self.headers)
```
Comment (sanfordstudent) [inline notesapi/v1/tests/test_views.py]: I see what's happening now. Django's `client.delete` takes args in the order `(path, data='', content_type='application/octet-stream', secure=False, **extra)` so `self.headers` is being passed as data to the request, not as headers.

Comment (yro) [reply notesapi/v1/tests/test_views.py]: Makes perfect sense. Will update.

**Comment by yro** [comment]
@sanfordstudent I believe I have updated this to reflect and assuage your concerns.

File: notesapi/v1/tests/test_views.py [hunk 3]
```diff
@@ -508,6 +507,40 @@ def test_pagination_page_start(self, page, start, previous_page, next_page):
...
+
+        # Delete for No User
+        del self.payload["user_id"]
+        response = self.client.delete(url, headers=self.headers, data=self.payload)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        # Delete for User 2 with no notes
+        self.payload["user_id"] = TEST_OTHER_USER
+        response = self.client.delete(url, headers=self.headers, data=self.payload)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```
Comment (doctoryes) [inline notesapi/v1/tests/test_views.py]: Nit: There's a lot of code paths tested in this single test. I would have preferred to see each case split out to a separate test, even though separate setUp/tearDown might need to be performed.

Comment (yro) [reply notesapi/v1/tests/test_views.py]: @doctoryes Sounds good - will do.

**Pull request merged**